src/toniarts/openkeeper/game/task/worker/DigTileTask.java

```diff
@@ -87,13 +87,14 @@ public String toString() {
 
     @Override
     public void executeTask(ICreatureController creature, float executionDuration) {
-
+//        if (creature.getMeleeAttack().execute()) {

```

Comment:
Hmm?

Reply:
When we execute the task, we must check whether it is possible to complete it. Specifically for DigTile, you need to check if the creature’s meele attack is recharged.

Reply:
Yeah, that needs a bit discussion that how we generally do that. Now I have defaulted all behavior to _measurement / per second_. The other way I guess would the to sync everything with the animations, that is what you are after here? I don't really have a clear preference which one is better and what the original game used.

Reply:
DigTile recharge == Melee attack recharge. Claim tile recharge == Some imp spell (or other attack) recharge